File: agent-gui/src-tauri/src/plugin_runtime/bootstrap.rs
use super::types::{PluginChannelEntry, PluginRegistryBootstrap};

const BOOTSTRAP_JSON: &str = include_str!("../../resources/plugin-registry-bootstrap.json");

pub fn load_bootstrap() -> Result<PluginRegistryBootstrap, String> {
    serde_json::from_str(BOOTSTRAP_JSON)
        .map_err(|e| format!("plugin-registry-bootstrap.json invalid: {e}"))
}

pub fn voice_channel_entry() -> Result<PluginChannelEntry, String> {
    load_bootstrap()?
        .offline_fallback_registry
        .plugins
        .get("voice-asr")
        .cloned()
        .ok_or_else(|| "bootstrap missing voice-asr channel entry".into())
}

File: agent-gui/electron/plugin-runtime/bootstrap.mjs
import { readFileSync } from "node:fs";
import { join } from "node:path";

/** @param {string} resourcesRoot */
export function loadBootstrap(resourcesRoot) {
  const path = join(resourcesRoot, "plugin-registry-bootstrap.json");
  const raw = readFileSync(path, "utf8");
  const data = JSON.parse(raw);
  return data;
}

/** @param {string} resourcesRoot */
export function voiceChannelEntry(resourcesRoot) {
  const bootstrap = loadBootstrap(resourcesRoot);
  const entry = bootstrap.offlineFallbackRegistry?.plugins?.["voice-asr"];
  if (!entry) {
    throw new Error("bootstrap missing voice-asr channel entry");
  }
  return entry;
}

/** @param {string} resourcesRoot */
export function loadEmbeddedVoiceChannel(resourcesRoot) {
  const raw = readFileSync(join(resourcesRoot, "voice-plugin-channel.json"), "utf8");
  return JSON.parse(raw);
}
